In [16]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

## Lendo a base de dados

In [17]:
df=pd.read_csv("Olympic_Athlete_Event_Results.csv")
df

edition  edition_id country_noc      sport  \
0       1908 Summer Olympics           5         ANZ  Athletics   
1       1908 Summer Olympics           5         ANZ  Athletics   
2       1908 Summer Olympics           5         ANZ  Athletics   
3       1908 Summer Olympics           5         ANZ  Athletics   
4       1908 Summer Olympics           5         ANZ  Athletics   
...                      ...         ...         ...        ...   
316829  2022 Winter Olympics          62         NED  Bobsleigh   
316830  2022 Winter Olympics          62         SVK  Bobsleigh   
316831  2022 Winter Olympics          62         KOR  Bobsleigh   
316832  2022 Winter Olympics          62         JAM  Bobsleigh   
316833  2022 Winter Olympics          62         UKR  Bobsleigh   

                  event  result_id                     athlete  athlete_id  \
0       100 metres, Men      56265             Ernest Hutcheon       64710   
1       400 metres, Men      56313                Henry Murray       64756   
2       800 metres, Men      56338               Harvey Sutton       64808   
3       800 metres, Men      56338                 Guy Haskins      922519   
4       800 metres, Men      56338                Joseph Lynch       64735   
...                 ...        ...                         ...         ...   
316829   Monobob, Women   19019671              Karlien Sleper      148662   
316830   Monobob, Women   19019671          Viktória Čerňanská      138683   
316831   Monobob, Women   19019671                  Kim Yu-Ran      137542   
316832   Monobob, Women   19019671  Jazmine Fenlator-Victorian      128682   
316833   Monobob, Women   19019671                Lidiya Hunko      149047   

              pos medal  isTeamSport  
0             DNS   NaN        False  
1             DNS   NaN        False  
2       3 h8 r1/2   NaN        False  
3             DNS   NaN        False  
4             DNS   NaN        False  
...           ...   ...          ...  
316829         16   NaN        False  
316830         17   NaN        False  
316831         18   NaN        False  
316832         19   NaN        False  
316833         20   NaN        False  

[316834 rows x 11 columns]

## Analisando os valores vazios

In [18]:
df.isnull().sum()

edition             0
edition_id          0
country_noc         0
sport               0
event               0
result_id           0
athlete             0
athlete_id          0
pos                 0
medal          272147
isTeamSport         0
dtype: int64

## Analisando o pais com mais medalhas

In [19]:
baseSemNUll=df.dropna()

paisesComMaisMedalhas=baseSemNUll["country_noc"].value_counts().reset_index()
paisesComMaisMedalhas

country_noc  count
0           USA   6294
1           URS   2546
2           GER   2444
3           GBR   2372
4           FRA   1976
..          ...    ...
149         GUA      1
150         IRQ      1
151         SAM      1
152         MON      1
153         AHO      1

[154 rows x 2 columns]

In [20]:
baseSemNUll
d=baseSemNUll["medal"].value_counts().reset_index()
d

medal  count
0    Gold  15072
1  Bronze  14939
2  Silver  14676

## Analisando o Melhor Atleta

In [21]:
def medal(value):
    if value =="Bronze":
        return 1
    elif value =="Silver":
        return 2
    else:
        return 3

baseSemNUll["medalPoint"]=baseSemNUll["medal"].apply(medal)

baseSemNUll
melhorAtleta=baseSemNUll[["athlete", "medalPoint"]]. groupby("athlete"). sum().sort_values(by="medalPoint", ascending=False).reset_index()
melhorAtleta


C:\Users\rianf\AppData\Local\Temp\ipykernel_14960\2983228215.py:9: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



athlete  medalPoint
0         Michael Phelps          77
1        Larisa Latynina          41
2          Marit Bjørgen          35
3         Jenny Thompson          34
4      Nikolay Andrianov          34
...                  ...         ...
31638      Peter Nordell           1
31639       Peter Nottet           1
31640          Gus Desch           1
31641      Gurjant Singh           1
31642       Ștefan Tudor           1

[31643 rows x 2 columns]

## Analisando os total de medalhas de ouro, prata e bronze do país com mais medalhas olimpicas

In [22]:
usa=baseSemNUll.loc[baseSemNUll["country_noc"]=="USA"]
usa["medal"].value_counts().reset_index()

medal  count
0    Gold   2942
1  Silver   1855
2  Bronze   1497

## Analisando a quantidade de olimpiadas

In [23]:
d=len(df["edition"].unique())
d


55

## Adicionando o Sexo dos atletas para uma coluna

In [24]:
def extrairGenero(texto):
    if "Men" in texto:
        return "Masculino"
    elif "Women" in texto:
        return "Feminino"
def medal(value):
    if value =="Bronze":
        return 1
    elif value =="Silver":
        return 2
    elif value=="Gold":
        return 3
    else:
        return 0

modalidade=df["event"]
df["sexo"]=df["event"].apply(extrairGenero)
df["medalPoint"]=df["medal"].apply(medal)
df.to_csv("Base_DashBoard", index=False)
df

edition  edition_id country_noc      sport  \
0       1908 Summer Olympics           5         ANZ  Athletics   
1       1908 Summer Olympics           5         ANZ  Athletics   
2       1908 Summer Olympics           5         ANZ  Athletics   
3       1908 Summer Olympics           5         ANZ  Athletics   
4       1908 Summer Olympics           5         ANZ  Athletics   
...                      ...         ...         ...        ...   
316829  2022 Winter Olympics          62         NED  Bobsleigh   
316830  2022 Winter Olympics          62         SVK  Bobsleigh   
316831  2022 Winter Olympics          62         KOR  Bobsleigh   
316832  2022 Winter Olympics          62         JAM  Bobsleigh   
316833  2022 Winter Olympics          62         UKR  Bobsleigh   

                  event  result_id                     athlete  athlete_id  \
0       100 metres, Men      56265             Ernest Hutcheon       64710   
1       400 metres, Men      56313                Henry Murray       64756   
2       800 metres, Men      56338               Harvey Sutton       64808   
3       800 metres, Men      56338                 Guy Haskins      922519   
4       800 metres, Men      56338                Joseph Lynch       64735   
...                 ...        ...                         ...         ...   
316829   Monobob, Women   19019671              Karlien Sleper      148662   
316830   Monobob, Women   19019671          Viktória Čerňanská      138683   
316831   Monobob, Women   19019671                  Kim Yu-Ran      137542   
316832   Monobob, Women   19019671  Jazmine Fenlator-Victorian      128682   
316833   Monobob, Women   19019671                Lidiya Hunko      149047   

              pos medal  isTeamSport       sexo  medalPoint  
0             DNS   NaN        False  Masculino           0  
1             DNS   NaN        False  Masculino           0  
2       3 h8 r1/2   NaN        False  Masculino           0  
3             DNS   NaN        False  Masculino           0  
4             DNS   NaN        False  Masculino           0  
...           ...   ...          ...        ...         ...  
316829         16   NaN        False   Feminino           0  
316830         17   NaN        False   Feminino           0  
316831         18   NaN        False   Feminino           0  
316832         19   NaN        False   Feminino           0  
316833         20   NaN        False   Feminino           0  

[316834 rows x 13 columns]

## Analisando o desempenho dos atletas em cada esporte

In [25]:
# Filtrar top 10 por sexo
def doisSexos(grupo):
    return len(grupo["sexo"].unique()) ==2

topFeminino=df[df["sexo"]=="Feminino"][["sport", "sexo", "medalPoint"]].groupby(["sport", "sexo"]).sum().sort_values(by="medalPoint", ascending=False).reset_index().head(10)
topMasculio=df[df["sexo"]=="Masculino"][["sport", "sexo", "medalPoint"]].groupby(["sport", "sexo"]).sum().sort_values(by="medalPoint", ascending=False).reset_index().head(10)
combine=pd.concat([topFeminino, topMasculio])
esportesAmbosSexos=combine.groupby("sport").filter(doisSexos)

df['sexo'] = df['sexo'].fillna(method='bfill')

# Gerar o gráfico
fig = px.bar(esportesAmbosSexos, x="sport", y="medalPoint", color="sexo")
fig.show()

C:\Users\rianf\AppData\Local\Temp\ipykernel_14960\3919581966.py:10: FutureWarning:

Series.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.



# Relatorio

## Melhores Paises das Olimpiadas
### -USA
### -URS
### -GER

## Melhores Atletas     
### 1	Michael Phelps	
### 2	Larisa Latynina	
### 3	Marit Bjørgen	
### 4	Jenny Thompson	

## Quantidades de medalhas dos USA

###	Gold	2942
###	Silver	1855
###	Bronze	1497

## O melhor desempenos baseados nas medalhas e seu valor nas Olimpíadas:

### Os homens foram os que tiveram os melhores resultados nas Olimpíadas

